### DISCUSSION TITLE
Two displays tft st7789, rp2040.

### DISCUSSION THREAD

**Discussion by MatSquikki** [Bodmer/TFT_eSPI#1727] [Q&A - General] [answered]
Hello
I am using your library with good results on raspberry rp2040.
I wanted to use two identical displays on the two different SPIs, but I didn't understand very well how to define my pins for the second display, and how to call up the second display in my code. Can you help me with examples?

as setup I use this successfully.`
////////////////////////////////////////////////////////////////////////////////////////////
// Interface
////////////////////////////////////////////////////////////////////////////////////////////
#define TFT_PARALLEL_16_BIT

////////////////////////////////////////////////////////////////////////////////////////////
// Display driver type
////////////////////////////////////////////////////////////////////////////////////////////
#define ST7789_DRIVER

////////////////////////////////////////////////////////////////////////////////////////////
// RP2040 pins used
////////////////////////////////////////////////////////////////////////////////////////////

//#define TFT_CS   -1  // Do not define, chip select control pin permanently connected to 0V

// These pins can be moved and are controlled directly by the library software
#define TFT_COUNT 2

//Definition of pin  TFTsx-----------------------------
#define TFT_CS         17
#define TFT_RST        21
#define TFT_DC         20
#define TFT_MOSI       19
#define TFT_SCLK       18

//Definition of pin   TFTdx----------------------------
//#define TFT1_CS         13
//#define TFT1_RST        15
//#define TFT1_DC         12
//#define TFT1_MOSI       11
//#define TFT1_SCLK       14

////////////////////////////////////////////////////////////////////////////////////////////
// Fonts to be available
////////////////////////////////////////////////////////////////////////////////////////////
#define LOAD_GLCD   // Font 1. Original Adafruit 8 pixel font needs ~1820 bytes in FLASH
#define LOAD_FONT2  // Font 2. Small 16 pixel high font, needs ~3534 bytes in FLASH, 96 characters
#define LOAD_FONT4  // Font 4. Medium 26 pixel high font, needs ~5848 bytes in FLASH, 96 characters
#define LOAD_FONT6  // Font 6. Large 48 pixel font, needs ~2666 bytes in FLASH, only characters 1234567890:-.apm
#define LOAD_FONT7  // Font 7. 7 segment 48 pixel font, needs ~2438 bytes in FLASH, only characters 1234567890:.
#define LOAD_FONT8  // Font 8. Large 75 pixel font needs ~3256 bytes in FLASH, only characters 1234567890:-.
#define LOAD_GFXFF  // FreeFonts. Include access to the 48 Adafruit_GFX free fonts FF1 to FF48 and custom fonts

#define SMOOTH_FONT

////////////////////////////////////////////////////////////////////////////////////////////
`

**Comment by Bodmer** [comment, marked as answer]
The library only supports one SPI bus at a time. You could put 2 displays on the same SPI bus if they are of identical type and manage the chip select in the sketch. See here:
https://www.youtube.com/watch?v=WL8Xs5D2VUA

**Comment by MatSquikki** [comment]
OK, great I was what I was looking for!
Examples to study on this ??

**Comment by Bodmer** [comment]
Link from video....
https://github.com/kitecraft/Multiple_TFT_Displays

**Comment by MatSquikki** [comment]
Hello,
I tried the "Basic_Double_TFT.ino" example but it doesn't work,
even changing pins and frequencies but my displays show nothing.

**Reply by Bodmer** [reply]
If I understand correctly you already have managed to get a single display working.  In that case go back to a single display and get that working. Next change the setup by commenting out the TFT_CS pin define thus:
// #define TFT_CS  xx // Where xx is the pin number you use
Now run Basic_Double_TFT.ino with the CS pin in that sketch to the xx and it should run. Next wire up the second TFT to the EXACT SAME pins and run again, now both displays should show the same image. Next move the CS signal from ONE display to another CS pin as defined in the sketch and it should all work fine.

**Reply by MatSquikki** [reply]
Hello
Nothing to do my displays do not turn on.
I attach files that I am using.
[Setup210_RPI2040_ST7789.txt](https://github.com/Bodmer/TFT_eSPI/files/8388653/Setup210_RPI2040_ST7789.txt)
[Basic_Double_TFT.txt](https://github.com/Bodmer/TFT_eSPI/files/8388654/Basic_Double_TFT.txt)
I noticed that if i don't comment the PIN_CS in the
"Setup210_RPI2040_ST7789.h" only one display lights up.

**Reply by Bodmer** [reply]
In your original post you say:
"I am using your library with good results on raspberry rp2040."

Now you say nothing works?

**Reply by MatSquikki** [reply]
if I use "Setup_1" the display turns on and works perfectly with any example of the library.
If, on the other hand, I try to use "Setup_2" where I comment the PIN_CS and define it in my code "Prova_1", the display does not show anything.
[Setup_1.txt](https://github.com/Bodmer/TFT_eSPI/files/8390989/Setup_1.txt)
[Setup_2 .txt](https://github.com/Bodmer/TFT_eSPI/files/8390990/Setup_2.txt)
[Prova_1.txt](https://github.com/Bodmer/TFT_eSPI/files/8390991/Prova_1.txt)
.

**Reply by Bodmer** [reply]
I tried dual display and it is working fine so we need to find out what is odd about your hardware and software setup.

You have defined a Raspberry Pi display type in the setup, do you have that display type?

Post a link to the display vendors page.

Run the test and diagnostic sketch "Read_User_Setup" and post the serial monitor output here. Detail which board package you are using  and the version number, you will find this in the Boards manager menu option.

The library does not support 16 bit displays and I see a commented out //#define TFT_PARALLEL_16_BIT. Are you using a fork of this library from somewhere else?

**Reply by MatSquikki** [reply]
Display used st7789 I attach photos.
Arduino ide version 1.8.13, board package Raspberry PI Pico / RP2040 By Earle F. Philhower version 1.12.0.
diagnostic sketch "Read_User_Setup" see File_1.
I had entered // # define TFT_PARALLEL_16_BIT because it intrigued me, but it is still not supported as a function and I had commented on it.
[FIle_1.txt](https://github.com/Bodmer/TFT_eSPI/files/8396128/FIle_1.txt)
![IMG_0319](https://user-images.githubusercontent.com/73945494/161228352-9cb880eb-781d-43ec-a67d-a7674fe628cb.JPG)

**Reply by Bodmer** [reply]
How are you powering the display?

**Reply by MatSquikki** [reply]
I am feeding the display with the Pico power supply (3.3V), I also tried an external power supply (3.3V).
